bouncer/tests/all_concurrent_tests.ts

```diff
@@ -51,6 +51,24 @@ async function runAllConcurrentTests() {
   await Promise.all(tests);
 
   await Promise.all([broadcastAborted.stop(), feeDeficitRefused.stop()]);
+
+  // Check that all Solana nonces are available
+  await using chainflip = await getChainflipApi();
+  const maxRetries = 5; // 30 seconds
+  for (let attempt = 0; attempt < maxRetries; attempt++) {
+    const availableNonces = (await chainflip.query.environment.solanaAvailableNonceAccounts())
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      .toJSON() as any[];
+    if (availableNonces.length === solanaNumberOfNonces) {
+      break;
+    } else if (attempt === maxRetries - 1) {
+      throw new Error(
+        `Unexpected number of available nonces: ${availableNonces.length}, expected ${solanaNumberOfNonces}`,
+      );
+    } else {
+      await sleep(6000);
+    }
+  }
```

Comment:
could we move this into a function, just to keep `runAllConcurrentTests` readable at a high level